service/src/main/java/com/epam/esm/servise/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.esm.servise.impl;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.User;
+import com.epam.esm.exception.IncorrectDataException;
+import com.epam.esm.exception.NoSuchEntityException;
+import com.epam.esm.mapper.impl.certificateMapper.CreateCertificateFromModelCertificateMapper;
+import com.epam.esm.mapper.impl.certificateMapper.ModelCertificateInOnlyCertificateMapper;
+import com.epam.esm.mapper.impl.certificateMapper.OnlyCertificateReadMapper;
+import com.epam.esm.mapper.impl.orderMapper.CreateOrderMapper;
+import com.epam.esm.mapper.impl.userMapper.CreateUserModelMapper;
+import com.epam.esm.mapper.impl.userMapper.UserModelReadMapper;
+import com.epam.esm.model.certificate.ModelCertificate;
+import com.epam.esm.model.order.CreateOrderModel;
+import com.epam.esm.model.user.ReadUserModel;
+import com.epam.esm.model.user.UserModel;
+import com.epam.esm.repository.impl.OrderRepositoryImpl;
+import com.epam.esm.repository.impl.UserRepositoryImpl;
+import com.epam.esm.servise.UserService;
+import com.epam.esm.util.messange.LanguageMassage;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service("UserService")
+@RequiredArgsConstructor
+
+public class UserServiceImpl implements UserService<User> {
+
+    private final UserRepositoryImpl repository;
```

Comment:
private final UserRepository repository;
 In variables it's better to define interface. Read about polymorphyism in OOP https://www.w3schools.com/java/java_polymorphism.asp

Reply:
made